code/acting/src/acting/trajectory_interpolation.py

```diff
@@ -0,0 +1,132 @@
+    old_len = vector_len(vector)
+    if old_len == 0:
+        return (0, 0)
+    scaled_vector = (vector[0] * new_len / old_len,
+                     vector[1] * new_len / old_len)
+    return scaled_vector
+
+
+def rotate_vector(vector: Tuple[float, float],
+                  angle_rad: float) -> Tuple[float, float]:
+    """Rotate the given vector by an angle"""
+    return (cos(angle_rad) * vector[0] - sin(angle_rad) * vector[1],
+            sin(angle_rad) * vector[0] + cos(angle_rad) * vector[1])
+
+
+def linear_interpolation(start: Tuple[float, float], end: Tuple[float, float],
+                         interval_m: float) -> List[Tuple[float, float]]:
+    """Interpolate linearly between the given start / end point
+    by putting points according to the interval specified."""
+
+    distance = euclid_dist(start, end)
+    vector = (end[0] - start[0], end[1] - start[1])
+
+    steps = max(1, floor(distance / interval_m))
+    exceeds_interval_cap = distance > interval_m
+    step_vector = (vector[0] / steps if exceeds_interval_cap else vector[0],
+                   vector[1] / steps if exceeds_interval_cap else vector[1])
+
+    lin_points = [(start[0], start[1])]
+    for i in range(1, steps):
+        lin_points.append(
+            (start[0] + step_vector[0] * i,
+             start[1] + step_vector[1] * i)
+        )
+
+    return lin_points
+
+
+'''
+def circular_interpolation(start: Tuple[float, float],
+                           end: Tuple[float, float],
+                           arc_radius: float) -> List[Tuple[float, float]]:
+    """Interpolate points between start / end point
+    on top of the circular arc given by the arc radius."""
+
+    step_size = 2.0
+    sign = -1 if arc_radius < 0 else 1
+    arc_radius = abs(arc_radius)
+
+    # determine the circular angle of the arc
+    angle = asin((euclid_dist(start, end) / 2) / arc_radius) * 2
+
+    # construct the mid-perpendicular of |start, end| to determine the center
+    conn_middle = ((start[0] + end[0]) / 2, (start[1] + end[1]) / 2)
+    center_off = sqrt(pow(arc_radius, 2) - pow(euclid_dist(start, end) / 2, 2))
+    mid_perpend = rotate_vector(points_to_vector(start, end), pi/2 * sign)
+    circle_center = add_vector(conn_middle,
+                              scale_vector(mid_perpend, center_offset)
+                              )
+
+    # partition the arc into steps (-> interpol. geometries)
+    arc_circumference = arc_radius * angle        # (r * 2 pi) * (angle / 2 pi)
+    num_steps = int(arc_circumference / step_size) + 1  # each step < step size
+
+    # compute the interpolated points on the circle arc
+    vec = points_to_vector(circle_center, start)
+    rot_angles = [angle * (i / num_steps) for i in range(num_steps+1)]
+    points = [add_vector(
+                        circle_center,
+                        rotate_vector(vec, rot * sign)
+                        )
+                        for rot in rot_angles]
+
+    return points
+'''
```

Comment:
It would be cleaner if you commented that part properly out, in my opinion a docstring might be misleading

Reply:
fixed in 93021b8